addons/stock/models/stock_orderpoint.py

```diff
@@ -172,6 +173,13 @@ def _compute_unwanted_replenish(self):
                 after_replenish_qty = orderpoint.product_id.with_context(company_id=orderpoint.company_id.id, location=orderpoint.location_id.id).virtual_available + orderpoint.qty_to_order
                 orderpoint.unwanted_replenish = float_compare(after_replenish_qty, orderpoint.product_max_qty, precision_rounding=orderpoint.product_uom.rounding) > 0
 
+    @api.depends('location_id')
+    def _compute_replenish_locations(self):
+        stock_locations = self.env['stock.location'].search([])
+        for orderpoint in self:
+            replenish_locations = stock_locations.filtered(lambda location: location.replenish_location)
```

Comment:
This line, like the `search` one, do not depend on the `orderpoint` variable. This is useless to do the `filtered` for each iteration